File: backend/app/schemas/comparativa.py
from pydantic import BaseModel
from typing import Optional


class ItemComparativa(BaseModel):
    nombre: str
    tipo: str  # "deuda" | "ahorro"
    monto: float
    tasa_anual: Optional[float] = None
    costo_o_rendimiento_mensual: float


class ComparativaResponse(BaseModel):
    deudas: list[ItemComparativa]
    ahorros: list[ItemComparativa]
    total_costo_deuda: float
    total_rendimiento_ahorro: float
    diferencia: float  # positivo = deuda > ahorro -> conviene pagar deuda
    recomendacion: str

File: frontend/src/types/comparativa.ts
export interface ItemComparativa {
  nombre: string
  tipo: 'deuda' | 'ahorro'
  monto: number
  tasa_anual: number | null
  costo_o_rendimiento_mensual: number
}

export interface ComparativaData {
  deudas: ItemComparativa[]
  ahorros: ItemComparativa[]
  total_costo_deuda: number
  total_rendimiento_ahorro: number
  diferencia: number
  recomendacion: string
}
